panels/examples/complexbin/animal.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/vimeo/dials/panels"
+)
+
+type animal struct {
+	Type  string // dog, lion, duck
+	Legs  int8
+	Group string // mammal, reptile, marsupial
+}
+
+type animals struct {
+	m map[string]animal
+}
+
+type animalTraits struct {
+	Name string
+}
+
+func (a *animals) Description(scPath []string) string {
+	return "Get a list of all the animals in the directory that match the criteria"
+}
+
+func (a *animals) ShortUsage(scPath []string) string {
+	return strings.Join(scPath, " ")
+}
+
+func (a *animals) LongUsage(scPath []string) string {
+	return strings.Join(scPath, " ") + "\n\tget all the animals"
+}
+
+func (a *animals) DefaultConfig() interface{}      { return &animalTraits{} }
+func (a *animals) SetupParams() panels.SetupParams { return panels.SetupParams{} }
+
+func (a *animals) Run(ctx context.Context, s *panels.Something) error {
+
+	rootDial := s.DialsPath[0]
+	c := rootDial.View().(*animal)
+	an := s.Dials.View().(*animalTraits)
+
+	foundMatch := false
+	for n, animal := range a.m {
+		if an.Name != "" && n != an.Name {
+			continue
+		}
+
+		if c.Group != "" && c.Group != animal.Group {
+			continue
+		}
+
+		if c.Legs != 0 && c.Legs != animal.Legs {
+			continue
+		}
+
+		if c.Type != "" && c.Type != animal.Type {
+			continue
+		}
+
+		foundMatch = true
+		s.W.Write([]byte(fmt.Sprintf("Name: %s\tType:%s\tLegs: %d\n", n, animal.Type, animal.Legs)))
```

Comment:
Oh, I missed this!
The `fmt.Fprintf` calls were fine, it's just that writing to a string-builder to then immediately write out its contents in an example like this was unnecessary complexity.
Let's switch this to:
```suggestion
		fmt.Fprintf(s.W,"Name: %s\tType:%s\tLegs: %d\n", n, animal.Type, animal.Legs)
```

Reply:
Good catch!